lib/rubocop/cop/lint/object_equality_override.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # Identifies `eql?` method definitions which do not
+      # check for argument class which might lead to runtime errors.
+      #
+      # @example
+      #
+      #   # bad
+      #   def eql?(other)
+      #     other.name == name
+      #   end
+      #
+      #   # bad
+      #   def eql?(other)
+      #     to_s == other.to_s
+      #   end
+      #
+      #   # good
+      #   def eql?(other)
+      #     other.is_a?(self.class) && other.name == name
+      #   end
+      #
+      #   # good
+      #   def eql?(other)
+      #     other.instance_of?(self.class) && to_s == other.to_s
+      #   end
+      class ObjectEqualityOverride < Base
+        MSG = 'Check against argument class within custom equality methods.'
```

Comment:
A thought for a more concise message:

```suggestion
        MSG = 'Check the class of `%<parameter_name>s` in `eql?`'
```

Reply:
Thanks.